{
	"schema_version": "1.6.0",
	"id": "UBUNTU-CVE-2020-35654",
	"modified": "2024-10-15T14:07:49Z",
	"published": "2021-01-12T09:15:00Z",
	"related": [
		"CVE-2020-35654",
		"USN-4697-1"
	],
	"details": "In Pillow before 8.1.0, TiffDecode has a heap-based buffer overflow when decoding crafted YCbCr files because of certain interpretation conflicts with LibTIFF in RGBA mode.",
	"severity": [
		{
			"type": "CVSS_V3",
			"score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H"
		}
	],
	"affected": [
		{
			"package": {
				"ecosystem": "Ubuntu:16.04:LTS",
				"name": "pillow",
				"purl": "pkg:deb/ubuntu/pillow?arch=src?distro=xenial"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "3.1.2-0ubuntu1.4"
						}
					]
				}
			],
			"versions": [
				"2.9.0-1",
				"3.0.0-1",
				"3.0.0-1build1",
				"3.1.0-1",
				"3.1.1-1",
				"3.1.2-0ubuntu1",
				"3.1.2-0ubuntu1.1",
				"3.1.2-0ubuntu1.3"
			],
			"ecosystem_specific": {
				"availability": "No subscription required",
				"binaries": [
					{
						"binary_name": "python-imaging",
						"binary_version": "3.1.2-0ubuntu1.4"
					},
					{
						"binary_name": "python-pil",
						"binary_version": "3.1.2-0ubuntu1.4"
					},
					{
						"binary_name": "python-pil-dbg",
						"binary_version": "3.1.2-0ubuntu1.4"
					},
					{
						"binary_name": "python-pil-doc",
						"binary_version": "3.1.2-0ubuntu1.4"
					},
					{
						"binary_name": "python-pil.imagetk",
						"binary_version": "3.1.2-0ubuntu1.4"
					},
					{
						"binary_name": "python-pil.imagetk-dbg",
						"binary_version": "3.1.2-0ubuntu1.4"
					},
					{
						"binary_name": "python3-pil",
						"binary_version": "3.1.2-0ubuntu1.4"
					},
					{
						"binary_name": "python3-pil-dbg",
						"binary_version": "3.1.2-0ubuntu1.4"
					},
					{
						"binary_name": "python3-pil.imagetk",
						"binary_version": "3.1.2-0ubuntu1.4"
					},
					{
						"binary_name": "python3-pil.imagetk-dbg",
						"binary_version": "3.1.2-0ubuntu1.4"
					}
				],
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2020/UBUNTU-CVE-2020-35654.json"
			}
		},
		{
			"package": {
				"ecosystem": "Ubuntu:18.04:LTS",
				"name": "pillow",
				"purl": "pkg:deb/ubuntu/pillow?arch=src?distro=bionic"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "5.1.0-1ubuntu0.3"
						}
					]
				}
			],
			"versions": [
				"4.1.1-3build2",
				"4.3.0-2ubuntu1",
				"5.0.0-1",
				"5.1.0-1",
				"5.1.0-1ubuntu0.2"
			],
			"ecosystem_specific": {
				"availability": "No subscription required",
				"binaries": [
					{
						"binary_name": "python-pil",
						"binary_version": "5.1.0-1ubuntu0.3"
					},
					{
						"binary_name": "python-pil-dbg",
						"binary_version": "5.1.0-1ubuntu0.3"
					},
					{
						"binary_name": "python-pil-doc",
						"binary_version": "5.1.0-1ubuntu0.3"
					},
					{
						"binary_name": "python-pil.imagetk",
						"binary_version": "5.1.0-1ubuntu0.3"
					},
					{
						"binary_name": "python-pil.imagetk-dbg",
						"binary_version": "5.1.0-1ubuntu0.3"
					},
					{
						"binary_name": "python3-pil",
						"binary_version": "5.1.0-1ubuntu0.3"
					},
					{
						"binary_name": "python3-pil-dbg",
						"binary_version": "5.1.0-1ubuntu0.3"
					},
					{
						"binary_name": "python3-pil.imagetk",
						"binary_version": "5.1.0-1ubuntu0.3"
					},
					{
						"binary_name": "python3-pil.imagetk-dbg",
						"binary_version": "5.1.0-1ubuntu0.3"
					}
				],
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2020/UBUNTU-CVE-2020-35654.json"
			}
		},
		{
			"package": {
				"ecosystem": "Ubuntu:20.04:LTS",
				"name": "pillow",
				"purl": "pkg:deb/ubuntu/pillow?arch=src?distro=focal"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "7.0.0-4ubuntu0.2"
						}
					]
				}
			],
			"versions": [
				"6.1.0-1",
				"6.1.0-1build1",
				"6.2.1-2",
				"7.0.0-4",
				"7.0.0-4build1",
				"7.0.0-4ubuntu0.1"
			],
			"ecosystem_specific": {
				"availability": "No subscription required",
				"binaries": [
					{
						"binary_name": "python-pil-doc",
						"binary_version": "7.0.0-4ubuntu0.2"
					},
					{
						"binary_name": "python3-pil",
						"binary_version": "7.0.0-4ubuntu0.2"
					},
					{
						"binary_name": "python3-pil-dbg",
						"binary_version": "7.0.0-4ubuntu0.2"
					},
					{
						"binary_name": "python3-pil.imagetk",
						"binary_version": "7.0.0-4ubuntu0.2"
					},
					{
						"binary_name": "python3-pil.imagetk-dbg",
						"binary_version": "7.0.0-4ubuntu0.2"
					}
				],
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2020/UBUNTU-CVE-2020-35654.json"
			}
		},
		{
			"package": {
				"ecosystem": "Ubuntu:20.04:LTS",
				"name": "pillow-python2",
				"purl": "pkg:deb/ubuntu/pillow-python2?arch=src?distro=focal"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "0"
						}
					]
				}
			],
			"versions": [
				"6.2.1-3~build1",
				"6.2.1-3"
			],
			"ecosystem_specific": {
				"ubuntu_priority": "medium"
			},
			"database_specific": {
				"source": "https://github.com/canonical/ubuntu-security-notices/blob/main/osv/cve/2020/UBUNTU-CVE-2020-35654.json"
			}
		}
	],
	"references": [
		{
			"type": "REPORT",
			"url": "https://ubuntu.com/security/CVE-2020-35654"
		},
		{
			"type": "REPORT",
			"url": "https://pillow.readthedocs.io/en/stable/releasenotes/index.html"
		},
		{
			"type": "REPORT",
			"url": "https://ubuntu.com/security/notices/USN-4697-1"
		},
		{
			"type": "REPORT",
			"url": "https://www.cve.org/CVERecord?id=CVE-2020-35654"
		}
	]
}
